src/lib/byProp.ts

```diff
@@ -23,7 +23,7 @@ export function concatByProp<T extends object, K extends ArrayKeys<T>>(
 ): T[K] {
   const all: any = [];
   objects.forEach((obj) => {
-    all.push(...obj[key]);
+    all.push(...(obj[key] as any));
```

Comment:
I don't understand why typescript stopped understanding my previous implementation. This is not beautiful but I don't want to spend too much time here